helix-core/src/main/java/org/apache/helix/task/JobRebalancer.java

```diff
@@ -704,4 +712,30 @@ private PartitionAssignment(String instance, String state) {
       _state = state;
     }
   }
+
+  /**
+   * Reset RUNNING/INIT tasks count in JobRebalancer
+   */
+  public static void resetActiveTaskCount(Collection<String> liveInstances, CurrentStateOutput currentStateOutput) {
+    // init participant map
+    for (String liveInstance : liveInstances) {
+      participantActiveTaskCount.put(liveInstance, 0);
+    }
+    // Active task == init and running tasks
+    fillActiveTaskCount(currentStateOutput.getPartitionCountWithPendingState(TaskConstants.STATE_MODEL_NAME,
```

Comment:
Yes, at least based on task state model.
And even there is not, it will be safer to add INIT check for pendingState.